openvdb/points/AttributeSet.h

```diff
@@ -69,7 +72,19 @@ class OPENVDB_API AttributeSet
 
     //////////
 
-    AttributeSet();
+    AttributeSet() = default;
+
+    /// Construct a new AttributeSet from the given Info and optionally steal
+    /// from an existing attribute set.
+    /// @param info the attribute info
+    /// @param arrayLength the desired length of the arrays in the new AttributeSet
+    /// @param lock an optional scoped registry lock to avoid contention
+    /// @param attributeSet an existing attribute set that can be stolen from
+    /// @note if provided, the attribute set in this constructor will be
+    /// cleared on completion of this constructor
+    AttributeSet(const Info& info, Index arrayLength,
+        const AttributeArray::ScopedRegistryLock* lock = nullptr,
+        AttributeSet* attributeSet = nullptr);
```

Comment:
I don't understand the purpose of the info class here. It seems that the same behaviour could be achieved with a bool `steal` flag. I'm not sure the benefit of consolidating the code justifies how convoluted this makes it (i.e. an info class which is expected to hold a descriptor also matched by a provided attribute set). You could, for example, achieve the same consolidation with a different function.

Reply:
It cannot be achieved with a new `steal` flag - the current constructor takes a const reference which is what is expected of a copy constructor. Even making the reference non-const would be confusing. There's no simple solution here in my opinion and adding another constructor with largely duplicate logic from the other two is not ideal. I don't even think the status quo is a good solution - having two similar, but subtly different pieces of logic for AttributeSet construction is not ideal in my opinion.

I consider the `Info` instance to be an implementation detail of the `AttributeSet(const AttributeSet&)` and `AttributeSet(const DescriptorPtr& descriptor)`, it's not required to know the internal mechanism for construction when using this API. I consider the `AttributeSet(const Info&,...)` constructor to be the advanced constructor, which you'd only use in certain specific cases.